libraries/resource_firewall_rule.rb

```diff
@@ -0,0 +1,32 @@
+class Chef
+  class Resource::FirewallRule < Resource
+    include Poise(Chef::Resource::Firewall)
+
+    IP_CIDR_VALID_REGEX = /\b(?:\d{1,3}\.){3}\d{1,3}\b(\/[0-3]?[0-9])?/
+
+    actions(:reject, :allow, :deny)
+
+    attribute :port, :kind_of => Integer
```

Comment:
Also `port`, `ports`, and `port_range` are redundant. Make a single `port` that can take an Int, Array, or Range.